samza-core/src/main/java/org/apache/samza/serializers/model/JsonJobModelMixIn.java

```diff
@@ -30,12 +30,12 @@
  */
 public abstract class JsonJobModelMixIn {
   @JsonCreator
-  public JsonJobModelMixIn(@JsonProperty("config") Config config, @JsonProperty("containers") Map<Integer, ContainerModel> containers) {
+  public JsonJobModelMixIn(@JsonProperty("config") Config config, @JsonProperty("containers") Map<String, ContainerModel> containers) {
```

Comment:
I think jackson decoder handles integer/string conversions correctly. I remember trying out a simple test before making this change. 
Let me try to add a test so that it uses this Mixin. That way, we can all sleep peacefully :)